contracts/interfaces/IServiceNodeContribution.sol

```diff
@@ -0,0 +1,343 @@
+
+    /// @notice Update the public keys and their proofs held in the contract.
+    ///
+    /// This can only be called prior to the operator contributing funds to the
+    /// contract or alternatively after they have called `reset`.
+    ///
+    /// @param newBLSPubkey The new 64 byte BLS public key for the node.
+    /// @param newBLSSig The new 128 byte BLS proof-of-posession signature that proves
+    /// the caller knows the secret component of `key`.
+    /// @param ed25519Pubkey The new 32 byte Ed25519 public key for the node.
+    /// @param ed25519Sig0 First 32 byte component of the signature for the Ed25519 key.
+    /// @param ed25519Sig1 Second 32 byte component of the signature for the Ed25519 key.
+    function updatePubkeys(BN256G1.G1Point memory newBLSPubkey,
+                           IServiceNodeRewards.BLSSignatureParams memory newBLSSig,
+                           uint256 ed25519Pubkey,
+                           uint256 ed25519Sig0,
+                           uint256 ed25519Sig1) external;
+
+    /// @notice Update the reservation slots for contributors held in this contract.
+    ///
+    /// This can only be called prior to the operator contributing funds to the
+    /// contract or alternatively after they have called `reset`.
+    ///
+    /// The list of reservations can be empty which will reset the contract,
+    /// deleting any reservation data that is currently held. The list cannot
+    /// specify more reservations than `maxContributors` or otherwise the
+    /// function reverts.
+    ///
+    /// @param reserved The new array of reserved contributors with their
+    /// proportion of stake they must fulfill in the node.
+    function updateReservedContributors(IServiceNodeRewards.ReservedContributor[] memory reserved) external;
+
+    /// @notice Update the beneficiary for the wallet/contributor that invokes
+    /// this call.
+    ///
+    /// If the caller is not a contributor in this contract, the contract
+    /// reverts.
+    function updateBeneficiary(address newBeneficiary) external;
+
+    /// @notice Contribute funds to the contract for the node run by
+    /// `operator`. The `amount` of SENT token must be at least the
+    /// `minimumContribution` or their amount specified in their reserved
+    /// contribution (if applicable) otherwise the contribution is reverted.
+    ///
+    /// Node registration parameters must be assigned prior to the operator
+    /// contribution or alternatively after `reset` is invoked.
+    ///
+    /// The operator must contribute their minimum contribution/reservation
+    /// before the public or reserved contributors can contribute to the node.
+    /// The minimum an operator can contribute is 25% of the staking requirement
+    /// regardless of having a reservation or not.
+    ///
+    /// @param amount The amount of SENT token to contribute to the contract.
+    function contributeFunds(uint256 amount, BeneficiaryData memory data) external;
```

Comment:
Could we get rid of the BeneficiaryData struct by just using an `address` argument instead (both here and other functions using it), and then just use `address(0)` to signify the "no separate beneficiary" value?  It's not a big deal, but makes it mildly nicer to interact with the contract than needing to pass a `[false, address(0)]` when you don't want a separate beneficiary.

Reply:
Using `address(0)` as "not specified" also matches how the vesting contract revoker address works.

Reply:
I thought about this, we could- I thought ... maybe there was some use case to set the benficiary to 0 if you wanted to demonstrably burn the funds. Perhaps that's an unfounded assumption

Reply:
I'd think sending to some obviously fake address would be a better way to do that than using the 0 address.

Reply:
I had patched this, now we reject the zero address with some unit tests as well to verify. Have a quick review of the latest commits and then you can merge if happy as I have another PR lined up that goes on top of this.